tokio/src/runtime/coop.rs

```diff
@@ -70,7 +76,10 @@ impl Budget {
 /// returns, the budget is reset to the value prior to calling the function.
 #[inline(always)]
 pub(crate) fn budget<R>(f: impl FnOnce() -> R) -> R {
-    with_budget(Budget::initial(), f)
+    with_budget(
+        context::get_init_budget().unwrap_or_else(|_| Budget::initial()),
+        f,
+    )
 }
```

Comment:
This introduces an extra thread-local access on every poll. @carllerche Thoughts? I think you said that you were working on reducing them.

Reply:
We should be removing TLS calls in the hot path at this point. It should be possible to build this without any more TLS lookups. 

I'm in the middle of a [big scheduler refactor](https://github.com/tokio-rs/tokio/compare/rt-threaded-rework). It will be easier to make this change after the refactor lands. Ideally in 2-3 weeks.

Reply:
For stable api,  `get_init_budget()` return value directly instead of accessing the TLS, see below
```rust
pub(super) fn get_init_budget() -> Result<coop::Budget, AccessError> {
    #[cfg(not(tokio_unstable))]
    {
        Ok(coop::Budget::default())
    }

    #[cfg(tokio_unstable)]
    {
        CONTEXT.try_with(|ctx| ctx.initial_budget.get())
    }
}

```